packages/tools/src/tools/SculptorTool.ts

```diff
@@ -0,0 +1,590 @@
+import { getEnabledElement } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { BaseTool } from './base';
+import { getAnnotations } from '../stateManagement';
+import {
+  EventTypes,
+  PublicToolProps,
+  ToolProps,
+  SVGDrawingHelper,
+  Annotations,
+} from '../types';
+import { point } from '../utilities/math';
+import { Events, ToolModes, AnnotationStyleStates } from '../enums';
+import { ToolGroupManager } from '../store';
+import { triggerAnnotationRenderForViewportIds } from '../utilities/triggerAnnotationRenderForViewportIds';
+import {
+  hideElementCursor,
+  resetElementCursor,
+} from '../cursors/elementCursor';
+import { StyleSpecifier } from '../types/AnnotationStyle';
+import { getStyleProperty } from '../stateManagement/annotation/config/helpers';
+import { triggerAnnotationModified } from '../stateManagement/annotation/helpers/state';
+import CircleSculptCursor from './SculptorTool/CircleSculptCursor';
+
+export type SculptData = {
+  mousePoint: Types.Point3;
+  mouseCanvasPoint: Types.Point2;
+  points: Array<Types.Point3>;
+  maxSpacing: number;
+  element: HTMLDivElement;
+};
+
+type CommonData = {
+  activeAnnotationUID: string | null;
+  viewportIdsToRender: any[];
+  isEditingOpenContour: boolean;
+  canvasLocation: Types.Point2 | undefined;
+};
+
+class SculptorTool extends BaseTool {
+  static toolName: string;
+  registeredShapes = new Map();
+  private isActive = false;
+  private selectedShape: string;
+  private commonData: CommonData = {
+    activeAnnotationUID: null,
+    viewportIdsToRender: [],
+    isEditingOpenContour: false,
+    canvasLocation: undefined,
+  };
+  private sculptData?: SculptData;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        minSpacing: 1,
+        referencedToolNames: [
+          'PlanarFreehandROI',
+          'PlanarFreehandContourSegmentationTool',
+        ],
+        toolShape: 'circle',
+        referencedToolName: 'PlanarFreehandROI',
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+    this.endCallback = this.endCallback.bind(this);
```

Comment:
This should be done by declaration instead - will comment int he right spot.  That avoids a separate bind.